blockly-ws-search/src/WorkspaceSearch.js

```diff
@@ -0,0 +1,655 @@
+      marker.setCurNode(curNode);
+    }
+  }
+
+  /**
+   * Closes the search bar.
+   */
+  close() {
+    this.setVisible_(false);
+    this.workspace_.markFocused();
+    this.clearBlocks();
+  }
+
+  /**
+   * Shows or hides the workspace search bar.
+   * @param {boolean} show Whether to set the search bar as visible.
+   * @private
+   */
+  setVisible_(show) {
+    this.htmlDiv_.style.display = show ? 'flex' : 'none';
+  }
+
+  /**
+   * Searches the workspace for the current search term and highlights matching
+   * blocks.
+   * @param {string} searchText The search text.
+   * @param {boolean=} preserveCurrent Whether to preserve the current block.
+   *    if it is included in the new matching blocks.
+   */
+  searchAndHighlight(searchText, preserveCurrent) {
+    const oldCurrentBlock = this.blocks_[this.currentBlockIndex_];
+    this.searchText_ = searchText.trim();
+    this.clearBlocks();
+    this.blocks_ = this.getMatchingBlocks_(
+        this.workspace_, this.searchText_, this.caseSensitive);
+    this.highlightSearchGroup_(this.blocks_);
+    let currentIdx = 0;
+    if (preserveCurrent) {
+      currentIdx = this.blocks_.indexOf(oldCurrentBlock);
+      currentIdx = currentIdx > -1 ? currentIdx : 0;
+    }
+    this.setCurrentBlock_(currentIdx);
+  }
+
+  /**
+   * Returns pool of blocks to search from.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace to get blocks from.
+   * @return {!Array.<!Blockly.BlockSvg>} The search pool of blocks to use.
+   * @private
+  */
+  getSearchPool_(workspace) {
+    const blocks = (
+        /** @type {!Array.<!Blockly.BlockSvg>} */
+        workspace.getAllBlocks(true));
+    return blocks.filter(block => {
+      // Filter out blocks contained inside of another collapsed block.
+      const surroundParent = block.getSurroundParent();
+      return !surroundParent || !surroundParent.isCollapsed();
+    });
+  }
+
+  /**
+   * Returns whether the given block matches the search text.
+   * @param {!Blockly.BlockSvg} block The block to check.
+   * @param {string} searchText The search text. Note if the search is case
+   *    insensitive, this will be passed already converted to lowercase letters.
+   * @param {boolean} caseSensitive Whether the search is caseSensitive.
+   * @return {boolean} Whether the block matches the search text.
+   * @private
+   */
+  isBlockMatch_(block, searchText, caseSensitive) {
+    let blockText = '';
+    if (block.isCollapsed()) {
+      // Search the whole string for collapsed blocks.
+      blockText = block.toString();
+    } else {
+      const topBlockText = [];
+      block.inputList.forEach(input => {
+        input.fieldRow.forEach(field => {
+          topBlockText.push(field.getText());
+        });
+      });
+      blockText = topBlockText.join(' ').trim();
+    }
+    if (!caseSensitive) {
+      blockText = blockText.toLowerCase();
+    }
+    return blockText.includes(searchText);
```

Comment:
string.``includes`` not available on IE11. https://caniuse.com/#feat=es6-string-includes
Either use ``indexOf`` or include the polyfill https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes#Polyfill

Reply:
done